packages/flutter_tools/test/crash_report_test.dart

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+import 'dart:math';
+
+import 'package:flutter_tools/src/base/context.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:test/test.dart';
+
+import 'package:flutter_tools/executable.dart' as tools;
+import 'package:flutter_tools/src/base/os.dart' as os;
+import 'package:flutter_tools/src/crash_reporting.dart';
+import 'package:flutter_tools/src/runner/flutter_command.dart';
+import 'src/context.dart';
+
+void main() {
+  group('crash reporting', () {
+    int testPort;
+
+    setUp(() async {
+      testPort = await os.findAvailablePort();
+      overrideBaseCrashUrlForTesting(Uri.parse('http://localhost:$testPort'));
+    });
+
+    tearDown(() {
+      resetBaseCrashUrlForTesting();
+    });
+
+    testUsingContext('should send crash reports', () async {
+      Completer<Null> reportReceived = new Completer<Null>();
+      String method;
+      Uri uri;
+
+      HttpServer server = await HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, testPort);
```

Comment:
Can you get this working with `LOOPBACK_IP_V6`?  Google's internal test infrastructure is very soon switching to IPV6-only, which'll prevent this test from running internally.